packages/contracts/src/manager/libraries/LibAppStorage.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {LibDiamond} from "./LibDiamond.sol";
+import "../access/AccessControl.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+
+bytes32 constant GOVERNANCE_TOKEN_MINTER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_MINTER_ROLE");
+bytes32 constant GOVERNANCE_TOKEN_BURNER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_BURNER_ROLE");
+bytes32 constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+bytes32 constant CREDIT_NFT_MANAGER_ROLE = keccak256("CREDIT_NFT_MANAGER_ROLE");
+bytes32 constant STAKING_MANAGER_ROLE = keccak256("STAKING_MANAGER_ROLE");
+bytes32 constant INCENTIVE_MANAGER_ROLE = keccak256("INCENTIVE_MANAGER");
+bytes32 constant GOVERNANCE_TOKEN_MANAGER_ROLE =
+    keccak256("GOVERNANCE_TOKEN_MANAGER_ROLE");
+
+struct AppStorage {
+    address twapOracleAddress;
```

Comment:
I would add a warning inside the comment here like
/* DO NOT CHANGE THE ORDER ONLY APPEND AT THE END*/

Reply:
> /* DO NOT CHANGE THE ORDER ONLY APPEND AT THE END*/

Yeah, that's absolutely true.
It's important to know that new state variable should be added only at the end of the struct when upgrading